update-format-crau/src/bin/test.rs

```diff
@@ -48,43 +55,92 @@ fn read_delta_update_header(f: &mut dyn Read) -> Result<DeltaUpdateFileHeader, B
     Ok(header)
 }
 
-fn main() -> Result<(), Box<dyn Error>> {
-    let path = std::env::args().nth(1).expect("missing path (second argument)");
-
-    let mut f = fs::File::open(path)?;
-    let header = read_delta_update_header(&mut f)?;
-
+// Take a file stream and DeltaUpdateFileHeader,
+// return a bytes slice of the actual signature data as well as its length.
+fn get_signatures_bytes<'a>(mut f: &'a File, header: &'a DeltaUpdateFileHeader) -> (Result<&'a [u8], Box<dyn Error>>, u64) {
     let manifest_bytes = {
         let mut buf = vec![0u8; header.manifest_size as usize];
-        f.read_exact(&mut buf)?;
+        f.read_exact(&mut buf).unwrap();
```

Comment:
Because I changed the return type of `get_signatures_bytes` to multiple values. Then it becomes impossible to propagate errors with `?`.
I think it is better to keep the return type as `Result` like before to keep `?` in all places.